src/models/HubbardAttractive/HubbardModelAttractive.jl

```diff
@@ -82,20 +81,35 @@ Hubbard model in the DQMC simulation.
 This isn't a performance critical method as it is only used once before the
 actual simulation.
 """
-function hopping_matrix(mc::DQMC, m::HubbardModelAttractive)
+function hopping_matrix(mc::DQMC, m::HubbardModelAttractive{L}) where {L<:AbstractCubicLattice}
     N = nsites(m)
     neighs = m.neighs # row = up, right, down, left; col = siteidx
 
     T = diagm(0 => fill(-m.mu, N))
 
     # Nearest neighbor hoppings
     @inbounds @views begin
-      for src in 1:N
-        for nb in 1:size(neighs,1)
-          trg = neighs[nb,src]
-          T[trg,src] += -m.t
+        for src in 1:N
+            for nb in 1:size(neighs,1)
+                trg = neighs[nb,src]
+                T[trg,src] += -m.t
+            end
+        end
+    end
+
+    return T
+end
+
+function hopping_matrix(mc::DQMC, m::HubbardModelAttractive{LT}) where {LT <: AbstractLattice}
+    N = nsites(m)
+
+    T = diagm(0 => fill(-m.mu, N))
+
+    # Nearest neighbor hoppings
+    @inbounds @views begin
+        for (src, trg) in neighbors(l, Val(false))
```

Comment:
See other comment: It would be better to construct a `neighbors_lookup_table` upfront.